### PR TITLE
`Modeling Exercise`: Add assessment of use case diagrams

### PR DESCRIPTION
**Pull request by terlan98**
This PR adds support for Use Case diagrams. Also, some parts of the code are refactored to further improve extensibility.

### CODE REVIEW

File: Themis/Models/UML/UMLRelationshipType.swift [hunk 1]
```diff
@@ -33,4 +33,16 @@ enum UMLRelationshipType: String, Decodable {
     case classDependency = "ClassDependency"
     case classAggregation = "ClassAggregation"
     case classComposition = "ClassComposition"
+    
+    /// The string that should be shown on the UML relationship to indicate the type of the element
+    var annotationTitle: String {
+        switch self {
+        case .useCaseExtend:
+            return "<<extend>>"
```
Comment (maximiliansoelch) [inline Themis/Models/UML/UMLRelationshipType.swift]: I think the much-less/much-more unicode sign looks better here:
```suggestion
            return "≪extend≫"
```

File: Themis/Models/UML/UMLRelationshipType.swift [hunk 2]
```diff
@@ -33,4 +33,16 @@ enum UMLRelationshipType: String, Decodable {
...
+        case .useCaseInclude:
+            return "<<include>>"
```
Comment (maximiliansoelch) [inline Themis/Models/UML/UMLRelationshipType.swift]: Same here:
```suggestion
            return "≪include≫"
```

**Review by maximiliansoelch — CHANGES_REQUESTED** [review]
Code looks good in general 👍
Just left one small suggestion

**Comment by terlan98** [comment]
@maximiliansoelch Thanks for the review! I implemented your suggestion.

File: Themis/Views/Assessment/Modeling Exercise/Diagram Renderers/Class Diagram/UMLClassDiagramRelationshipRenderer.swift [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+//  Themis
+//
+//  Created by Tarlan Ismayilsoy on 27.07.23.
+//
+
+import SwiftUI
+import Common
+
+// swiftlint:disable:next type_body_length
+struct UMLClassDiagramRelationshipRenderer: UMLDiagramRenderer {
```
Comment (terlan98) [inline Themis/Views/Assessment/Modeling Exercise/Diagram Renderers/Class Diagram/UMLClassDiagramRelationshipRenderer.swift:12]: Not sure why the SwiftLint action is complaining here. Removing this rule causes a `type_body_length` error 🤷‍♂️

**Review by maximiliansoelch — APPROVED** [review]
Thanks for implementing the change!
Approved 👍

**Pull request merged**